sansa-ml-common/src/main/scala/net/sansa_stack/ml/common/outliers/anomalyDetection/Utils.scala

```diff
@@ -0,0 +1,67 @@
+package net.sansa_stack.ml.common.outliers.anomalyDetection
+
+import scala.collection.mutable
+import scala.collection.mutable.HashSet
+
+import org.apache.jena.graph.Node
+
+object Utils extends Serializable {
+
+  def isNumeric(x: String): Boolean =
+    {
+      if (x.contains("^")) {
+        val c = x.indexOf('^')
+        val subject = x.substring(1, c - 1)
+
+        if (isAllDigits(subject)) true
+        else false
+      } else false
+    }
+
+  def isAllDigits(x: String): Boolean = {
+    var found = false
+    for (ch <- x) {
+      if (ch.isDigit || ch == '.') {
+        found = true
+      } else if (ch.isLetter) {
+
+        found = false
+      }
+    }
+
+    found
+  }
+
+  def searchedge(x: String, y: List[String]): Boolean = {
+    if (x.contains("^")) {
+      val c = x.indexOf('^')
+      val subject = x.substring(c + 2)
+      y.contains(subject)
+    } else false
+  }
+
+  def getLocalName(x: Node): String = {
```

Comment:
Not sure why do you need to re-implement `getLocalName` method here, it is already on the Jena Model : https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/rdf/model/Resource.html#getLocalName-- and you can just call it when `f.getSubject.getLocalName`, or you tried that one and didn't work?

Reply:
Because if the name has brackets ( "(")  in it, then it was showing an empty result. For example, "<http://dbpedia.org/resource/David_Sánchez_(tennis)>", getlocalName returns no results.

Reply:
Really! Then maybe we should fire an issue to https://github.com/apache/jena for that :) . We can keep this as it is for now, but we should also double check this with the Jena developers.